app/controlplane/pkg/data/attestationstate.go

```diff
@@ -48,26 +51,69 @@ func (r *AttestationStateRepo) Initialized(ctx context.Context, runID uuid.UUID)
 	return exists, nil
 }
 
-func (r *AttestationStateRepo) Save(ctx context.Context, runID uuid.UUID, state []byte) error {
-	err := r.data.DB.WorkflowRun.UpdateOneID(runID).SetAttestationState(state).Exec(ctx)
+// baseDigest, when provided will be used to check that it matches the digest of the state currently in the DB
+// if the digests do not match, the state has been modified and the caller should retry
+func (r *AttestationStateRepo) Save(ctx context.Context, runID uuid.UUID, state []byte, baseDigest string) (err error) {
+	tx, err := r.data.DB.Tx(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create transaction: %w", err)
+	}
+
+	defer func() {
+		// Unblock the row if there was an error
+		if err != nil {
+			_ = tx.Rollback()
+		}
+	}()
+
+	// compared the provided digest with the digest of the state in the DB
+	// TODO: make digest check mandatory on updates
```

Comment:
I guess we can do this when all clients have been updated, right?

Reply:
Correct, for now it's not required.